exercicios/para-casa/Geice-SousaPinho/clinicaVeterinaria.js

```diff
@@ -0,0 +1,76 @@
+function Gatos( nome, idade, cor, castrado){ // com letra maiúscula por boa prática par mostrar que é um objeto
+    this.nome = nome;
+    this.idade = idade + ' meses'; //em meses
+    this.cor = cor;
+    this.castrado = castrado; // boolean
+
+    this.miar = ()=>{
+        return `O gato ${this.nome} mia.`
+    }
+
+    this.historico = new HistoricoMedico();
+}
+
+function Cachorros( nome, idade, cor, castrado, raca){ 
+    this.nome = nome;
+    this.idade = idade + ' meses';
+    this.cor = cor;
+    this.castrado = castrado; 
+    this.raca = raca;
+
+    this.latir = function latir(){
+        return `O cachorro ${this.nome} late.` //podia ser só o console apenas para imprimir no terminal ou só return para armazenar na memoria
+    }
+
+    this.historico = new HistoricoMedico();
+}
+
+function AnimaisExoticos( nome, idade, cor, especie){ 
+    this.nome = nome;
+    this.idade = idade + ' meses';
+    this.cor = cor;
+    this.especie = especie; 
+
+    this.ehAdulto = function(){
+        // poderia usar um while?
```

Comment:
Nesse caso, while não daria para ser aplicado pois não é um loop. 
Eu fiz de uma outra forma e acho que ficou interessante tbm, caso queira ver.


AnimalExotico.prototype.ehAdulto = function () {
	if (
		(this.especie === "hamster" && this.idade > 2) ||
		(especie === "papagaio" && idade > 36) ||
		(especie === "hamster" && idade > 18)
	) {
		return true;
	} else {
		return false;
	}
};

Reply:
eu coloquei while, mas pensei no switch hahahaha
peguei esse exemplo do mdn, dá pra aplicar assim tbm